BungieFrameWork/BFW_Source/BFW_LocalInput/Platform_SDL/BFW_LI_Platform_SDL.c

```diff
@@ -0,0 +1,630 @@
+{
+	UUtUns32 oni_modifiers = 0;
+	if (sdl_modifiers & KMOD_LSHIFT)
+		oni_modifiers |= LIcKeyState_LShiftDown;
+	if (sdl_modifiers & KMOD_RSHIFT)
+		oni_modifiers |= LIcKeyState_RShiftDown;
+	if (sdl_modifiers & KMOD_LCTRL)
+		oni_modifiers |= LIcKeyState_LCommandDown;
+	if (sdl_modifiers & KMOD_RCTRL)
+		oni_modifiers |= LIcKeyState_RCommandDown;
+	if (sdl_modifiers & KMOD_LALT)
+		oni_modifiers |= LIcKeyState_LAltDown;
+	if (sdl_modifiers & KMOD_RALT)
+		oni_modifiers |= LIcKeyState_RAltDown;
+	return oni_modifiers;
+}
+
+static SDL_Keymod
+sdl_to_kmod(SDL_Keycode sym)
+{
+	switch (sym)
+	{
+		case SDLK_LCTRL:  return KMOD_LCTRL;
+		case SDLK_RCTRL:  return KMOD_RCTRL;
+		case SDLK_LSHIFT: return KMOD_LSHIFT;
+		case SDLK_RSHIFT: return KMOD_RSHIFT;
+		case SDLK_LALT:   return KMOD_LALT;
+		case SDLK_RALT:   return KMOD_RALT;
+		default:          return KMOD_NONE;
+	}
+}
+
+UUtUns32
+LIrPlatform_InputEvent_InterpretModifiers(
+	LItInputEventType	inEventType,
+	UUtUns32			inModifiers)
+{
+	(void)inEventType;
+	// already translated in LIrPlatform_Update()
+	return inModifiers;
+}
+
+// ----------------------------------------------------------------------
+void
+LIrPlatform_InputEvent_GetMouse(
+	LItMode				inMode,
+	LItInputEvent		*outInputEvent)
+{
+	// set up the outInputEvent
+	outInputEvent->type			= LIcInputEvent_None;
+	outInputEvent->where.x		= 0;
+	outInputEvent->where.y		= 0;
+	outInputEvent->key			= 0;
+	outInputEvent->modifiers	= 0;
+
+	if (inMode == LIcMode_Normal)
+	{
+		int x, y;
+		
+		Uint32 buttons = SDL_GetMouseState(&x, &y);
+		outInputEvent->where.x = (UUtInt16)x;
+		outInputEvent->where.y = (UUtInt16)y;
+		// Mac checks Shift, left and right mouse buttons
+		if (LIrPlatform_TestKey(LIcKeyCode_LeftShift, inMode))
+			outInputEvent->modifiers |= LIcMouseState_LShiftDown;
+		if (buttons & SDL_BUTTON_LMASK)
+			outInputEvent->modifiers |= LIcMouseState_LButtonDown;
+		if (buttons & SDL_BUTTON_RMASK)
+			outInputEvent->modifiers |= LIcMouseState_RButtonDown;
+	}
+}
+
+// ======================================================================
+#if 0
+#pragma mark -
+#endif
+// ======================================================================
+// ----------------------------------------------------------------------
+void
+LIrPlatform_Mode_Set(
+	LItMode				inMode)
+{
+	int x = 0, y = 0;
+	if (inMode == LIcMode_Normal)
+	{
+		SDL_SetWindowGrab(LIgWindow, SDL_FALSE);
+		SDL_SetRelativeMouseMode(SDL_FALSE);
+	}
+	else
+	{
+		SDL_SetWindowGrab(LIgWindow, SDL_TRUE);
+		SDL_SetRelativeMouseMode(SDL_TRUE);
+		// reset mouse movement
+		SDL_GetRelativeMouseState(&x, &y);
+	}
+}
+
+// ======================================================================
+#if 0
+#pragma mark -
+#endif
+// ======================================================================
+// ----------------------------------------------------------------------
+UUtError
+LIrPlatform_Initialize(
+	UUtAppInstance		inInstance,
+	UUtWindow			inWindow)
+{
```

Comment:
I'm thinking, perhaps we should split SDL2 usage more.
As with the rendering, we should be able to switch between native input and SDL in put.
That means I would add multiple cmake options: 
- an option to switch between native input and sdl2 input (I think this needs the SDL2 events subsystem)
- an option to switch between native rendering and sdl2 rendering (with both using opengl)

It would allow us to compare input behavior without needing to worry about differences in rendering behavior.